sdk/java/pulumi/src/test/java/com/pulumi/serialization/internal/ConverterTests.java

```diff
@@ -226,7 +227,8 @@ void testNullInPreviewProducesFalseKnown() {
             DeploymentTests.DeploymentMockBuilder.builder()
                     .setMocks(new MocksTest.MyMocks())
                     .setOptions(new TestOptions(true))
-                    .setMockGlobalInstance();
+                    .deploymentFactory(MockDeployment::new)
```

Comment:
we choose to mock the deployment in cases where we know we don't need it, since it requires knowledge about internals I've decided this should not be the default anymore, so this is done explicitly now